File: be/user_be_java/src/main/java/com/user/common/response/ApiResponse.java
package com.user.common.response;

import lombok.Data;

import java.util.UUID;

/**
 * 统一 API 响应格式
 */
@Data
public class ApiResponse<T> {
    private Integer code;
    private String message;
    private T data;
    private String traceId;

    public ApiResponse(Integer code, String message, T data) {
        this.code = code;
        this.message = message;
        this.data = data;
        this.traceId = UUID.randomUUID().toString();
    }

    /**
     * 成功响应
     */
    public static <T> ApiResponse<T> success(T data) {
        return new ApiResponse<>(0, "success", data);
    }

    /**
     * 成功响应（带自定义消息）
     */
    public static <T> ApiResponse<T> success(T data, String message) {
        return new ApiResponse<>(0, message, data);
    }

    /**
     * 错误响应
     */
    public static <T> ApiResponse<T> error(Integer code, String message) {
        return new ApiResponse<>(code, message, null);
    }
}

File: be/user_be_go/pkg/response/api_response.go
package response

import (
	"github.com/gin-gonic/gin"
	"github.com/google/uuid"
)

// ApiResponse 统一响应格式
type ApiResponse struct {
	Code    int         `json:"code"`
	Message string      `json:"message"`
	Data    interface{} `json:"data"`
	TraceID string      `json:"traceId"`
}

// Success 成功响应
func Success(c *gin.Context, data interface{}, message ...string) {
	msg := "success"
	if len(message) > 0 && message[0] != "" {
		msg = message[0]
	}
	c.JSON(200, ApiResponse{
		Code:    0,
		Message: msg,
		Data:    data,
		TraceID: uuid.New().String(),
	})
}

// Error 错误响应
func Error(c *gin.Context, code int, message string) {
	c.JSON(code, ApiResponse{
		Code:    code,
		Message: message,
		Data:    nil,
		TraceID: uuid.New().String(),
	})
}

// Fail 失败响应（用于构造响应对象，不直接发送）
func Fail(code int, message string) ApiResponse {
	return ApiResponse{
		Code:    code,
		Message: message,
		Data:    nil,
		TraceID: uuid.New().String(),
	}
}

